fs-index/src/watch.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{ResourceIndex, ARK_FOLDER};
+use anyhow::Result;
+use futures::{
+    channel::mpsc::{channel, Receiver},
+    SinkExt, StreamExt,
+};
+use log::info;
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use std::{fs, path::Path};
+
+/// Watch the root path for changes and update the index
+pub async fn watch_index<P: AsRef<Path>>(root_path: P) -> Result<()> {
+    log::debug!(
+        "Attempting to watch index at root path: {:?}",
+        root_path.as_ref()
+    );
+
+    let root_path = fs::canonicalize(root_path.as_ref())?;
+    let mut index = ResourceIndex::provide(&root_path)?;
+
+    let (mut watcher, mut rx) = async_watcher()?;
+    info!("Watching directory: {:?}", root_path);
+    let config = Config::default();
+    watcher.configure(config)?;
+    watcher.watch(root_path.as_ref(), RecursiveMode::Recursive)?;
+    info!("Started watcher with config: \n\t{:?}", config);
+
+    while let Some(res) = rx.next().await {
+        match res {
+            Ok(event) => {
+                // If the event is a change in .ark folder, ignore it
+                let ark_folder = root_path.join(ARK_FOLDER);
+                if event
+                    .paths
+                    .iter()
+                    .any(|p| p.starts_with(&ark_folder))
+                {
+                    continue;
+                }
```

Comment:
I would expect something like this:
```
let paths = event.paths.filter(|p| !p.start_with(&ark_folder));

index.update(paths)
```

Reply:
So, we need new `fn update(path: Vec<Path>)` for this. It would be also called by `update_all`.

Reply:
Exactly. I'll implement a helper function to update the indexes of multiple directories with one call. This will be part of the next phase, which will be handled in a separate PR. 

Specifically, we'll require this function for the 
```sh
ark-cli watch --all
```
command